File: floyd_triangle.js
function floyd(numRows) {
  var rowLen, x, i = 0
  for (rowLen = 1; rowLen <= numRows; rowLen++) {
    var row = new Array()
    for (x = 0; x < rowLen; x++) {
      i += 1
      row.push(i)
    }
    console.log(row.join(' '))
  }
}

floyd(5)

File: floyd_triangle.hs
import Data.List (intercalate)

main = putStrLn $ format $ floyd 5
  where
    format rows = intercalate "\n" $ [ intercalate " " $ map show $ row | row <- rows ]

floyd numRows = take numRows $ loop [1..] 1
  where
    loop xs rowLen = row : (loop rest (rowLen + 1))
      where (row, rest) = splitAt rowLen xs
